source/gfoidl.DataCompression/Compression/DeadBandCompression.cs

```diff
@@ -200,6 +208,53 @@ public override bool MoveNext()
                 }
             }
             //-----------------------------------------------------------------
+            /// <summary>
+            /// Returns an array of the compressed <see cref="DataPoint" />s.
+            /// </summary>
+            /// <returns>An array of the compressed <see cref="DataPoint" />s.</returns>
+            public override DataPoint[] ToArray()
+            {
+                IEnumerator<DataPoint> enumerator = _source.GetEnumerator();
+
+                if (!enumerator.MoveNext())
+                    return Array.Empty<DataPoint>();
+
+                DataPoint snapShot = enumerator.Current;
+                _lastArchived      = snapShot;
+                DataPoint incoming = snapShot;          // sentinel, nullable would be possible but to much work around
+
+                var arrayBuilder = new ArrayBuilder<DataPoint>(true);
+                arrayBuilder.Add(snapShot);
+                this.GetBounding(snapShot);
+
+                while (enumerator.MoveNext())
+                {
+                    incoming        = enumerator.Current;
+                    ref var archive = ref this.IsPointToArchive(incoming);
+
+                    if (!archive.Archive)
+                    {
+                        snapShot = incoming;
+                        continue;
+                    }
+
+                    if (!archive.MaxDelta)
+                        arrayBuilder.Add(snapShot);
+
+                    arrayBuilder.Add(incoming);
+
+                    snapShot      = incoming;
+                    _lastArchived = incoming;
+
+                    if (!_archive.MaxDelta) this.GetBounding(snapShot);
```

Comment:
This is `UpdatePoints` kind of inlined -- reuse existing `UpdatePoints` (signature change necessary to accept `snaptShot`).